File: CodeWars/Python/simple_pig_latin.py
import re
def pig_it(text):
    def swap(w):
        if re.search('\W', w) == None:
            return w[1:] + str(list(w).pop(0))+'ay'
        return w
    return ' '.join(list(map(swap, text.split(' '))))
    
print(pig_it('hi fly guy !'))

# better answer
# def pig_it(text):
#     return ' '.join(w[1:]+w[0]+'ay' if w.isalpha() else w for w in text.split(' '))

File: CodeWars/JavaScript/SimplePigLatin.js
// https://www.codewars.com/kata/520b9d2ad5c005041100000f/train/javascript
function pigIt(str){
    return str.split(' ').map((w) => {
        return w.search(/\W/) < 0 ? w.split('').slice(1).concat(w[0]+'ay').join('') : w
    }).join(' ')
}

console.log(pigIt('hi fly guy !'))

// better answer
// function pigIt(str){
//     return str.replace(/(\w)(\w*)(\s|$)/g, "\$2\$1ay\$3")
// }